### DISCUSSION TITLE
Not able to set content_disposition for object.presigned_url

### DISCUSSION THREAD

**Discussion by bonniesimon** [aws/aws-sdk-ruby#2964] [Q&A]
I can't seem to set content_disposition for object.presigned_url

```ruby
    s3_object = Aws::S3::Object.new(
        BUCKET_NAME,
        s3_key,
        { client: s3_client }
      )

    s3_object.presigned_url(
      :upload_part,
      part_number:,
      upload_id:,
      key: s3_key,
     content_disposition: "attachment"
    )
```

The above will throw an exception `{"error":"unexpected value at params[:content_disposition]"}`

When stepping in through the debugger, this is coming from https://github.com/aws/aws-sdk-ruby/blob/6743619655ecfa1db72af71aade6e9235163522a/gems/aws-sdk-s3/lib/aws-sdk-s3/presigner.rb#L148

**Comment by mullermp** [comment]
Thanks for opening up a discussion. I think the member you are looking for is called `response_content_disposition`

https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Types/GetObjectRequest.html#response_content_disposition-instance_method

**Comment by bonniesimon** [comment]
I did try setting that. It gave me the same error.
<img width="675" alt="image" src="https://github.com/aws/aws-sdk-ruby/assets/49012815/232ae901-2a31-43f3-ae76-3690914c4061">


Note that here I'm trying to upload a file in multipart manner. Hence, I create presigned_url for each individual parts. And I'm unable to set the content disposition of the whole content.

**Reply by mullermp** [reply]
Ah! I see. Looking at upload_part, there doesn't appear to be content disposition. It looks like you need to set that on create_multipart_upload when you first get your upload_id.
